lib/pbench/cli/server/shell.py

```diff
@@ -194,6 +195,32 @@ def main():
         logger.info("Created database {}", db_uri)
     Database.init_db(server_config, logger)
 
+    # Add the Pbench Server binaries to the PATH.
+    os.environ["PATH"] = ":".join([str(server_config.BINDIR), os.environ["PATH"]])
+
+    crontab_f = Path(crontab_dir) / "crontab"
+    if not crontab_f.exists():
+        # Create the crontab file from the server configuration.
+        cp = subprocess.run(
+            ["pbench-create-crontab", crontab_dir],
+            cwd=server_config.log_dir,
+        )
+        if cp.returncode != 0:
+            logger.error(
+                "Failed to create crontab file from configuration: {}", cp.returncode
+            )
+            crontab_f.unlink(missing_ok=True)
+            sys.exit(1)
+
+        # Install the created crontab file.
+        cp = subprocess.run(
+            ["crontab", f"{crontab_dir}/crontab"], cwd=server_config.log_dir
+        )
+        if cp.returncode != 0:
+            logger.error("Failed to install crontab file: {}", cp.returncode)
+            crontab_f.unlink(missing_ok=True)
+            sys.exit(1)
```

Comment:
Alternately,
```
    crontab_f = Path(crontab_dir) / "crontab"
    if not crontab_f.exists():
        # Add the Pbench Server binaries to the PATH.
        os.environ["PATH"] = ":".join([str(server_config.BINDIR), os.environ["PATH"]])

        # Create the crontab file from the server configuration.
        cp = subprocess.run(
            ["pbench-create-crontab", crontab_dir],
            cwd=server_config.log_dir,
        )
        if cp.returncode != 0:
            logger.error(
                "Failed to create crontab file from configuration: {}", cp.returncode
            )
        else:
            # Install the created crontab file.
            cp = subprocess.run(
                ["crontab", f"{crontab_dir}/crontab"], cwd=server_config.log_dir
            )
            if cp.returncode != 0:
                logger.error("Failed to install crontab file: {}", cp.returncode)

        if cp.returncode != 0:
            crontab_f.unlink(missing_ok=True)
            sys.exit(1)
```
That is, if either of the operations fails, then we remove the crontab file and exit, but we only do it in one place (and, we only modify the `PATH` if we're gonna try the operations).

FWIW, it would be better if we delegated most of this code to a subroutine, which would allow the flow to return prematurely in the event of an error; we could pass `crontab_f` in so that the caller can unlink it if the subroutine returns a failure.

Reply:
And, I'll just pre-resolve this, in case you're not interested....

Reply:
Since there were unresolved comments and you went through the trouble to derive the logic which gave me another idea, why not!